core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -197,7 +197,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
-    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+    public SuggestionContext build(QueryShardContext context) throws IOException {
+        SuggestionContext suggestionContext = innerBuild(context);
+        // copy over common settings to each suggestion builder
+        populateCommonFields(context.getMapperService(), suggestionContext);
```

Comment:
I think the fact that we `populateCommonFields` after `innerBuild` can get messy :). In `innerBuild` we are assuming that the `suggestionContext` will not have their common fields populated but in reality, there is nothing stopping us from wrongly accessing the common fields in the `innerBuild`. In fact, currently `PhraseSuggestionBuilder.innerBuild` does use the common `getField` and `getAnalyzer` (it does get set before though) before the fields are populated.